### PR TITLE
Adds fire ants as a chem

### PR DESCRIPTION
**Pull request by Momo8289**
## About The Pull Request
Adds fire ants as a chem. They work pretty much exactly the same as normal ants, but they deal roughly 2x damage. Also mildly cleans up whatever ant code I touched
## Why It's Good For The Game
Doesn't really make sense that when you scoop up an anthill of fire ants, you get a beaker full of normal ants. It's honestly an injustice. Also, new flavours for the bartender and chef to work with are always nice.
## Changelog
:cl:
add: Fire ants can now be scooped up and used as a chem like normal space ants
/:cl:

### CODE REVIEW

**Comment by Dawnseer** [comment]
Finally, my ant based cuisine can have ant varieties, spicy, and very spicy!

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 1]
```diff
@@ -2892,15 +2898,15 @@
...
 	. = ..()
-	victim.adjustBruteLoss(max(0.1, round((ant_damage * 0.025),0.1))) //Scales with time. Roughly 32 brute with 100u.
+	victim.adjustBruteLoss(max(0.1, round((ant_damage * ant_damage_coeff),0.1))) //Scales with time. Roughly 32 brute with 100u.
 	ant_damage++
 	if(ant_damage < 5) // Makes ant food a little more appetizing, since you won't be screaming as much.
 		return
 	if(SPT_PROB(5, seconds_per_tick))
 		if(SPT_PROB(5, seconds_per_tick)) //Super rare statement
-			victim.say("AUGH NO NOT THE ANTS! NOT THE ANTS! AAAAUUGH THEY'RE IN MY EYES! MY EYES! AUUGH!!", forced = /datum/reagent/ants)
+			victim.say("AUGH NO NOT THE ANTS! NOT THE ANTS! AAAAUUGH THEY'RE IN MY EYES! MY EYES! AUUGH!!", forced = src.type)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:2907]: why src?

```suggestion
			victim.say("AUGH NO NOT THE ANTS! NOT THE ANTS! AAAAUUGH THEY'RE IN MY EYES! MY EYES! AUUGH!!", forced = type)
```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 2]
```diff
@@ -2892,15 +2898,15 @@
...
 		else
-			victim.say(pick(ant_screams), forced = /datum/reagent/ants)
+			victim.say(pick(ant_screams), forced = src.type)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:2909]: source? source? can I see your source? cite your source.

```suggestion
			victim.say(pick(ant_screams), forced = type)
```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 3]
```diff
@@ -2909,15 +2915,15 @@
 /datum/reagent/ants/on_mob_end_metabolize(mob/living/living_anthill)
 	. = ..()
 	ant_damage = 0
-	to_chat(living_anthill, "<span class='notice'>You feel like the last of the ants are out of your system.</span>")
+	to_chat(living_anthill, span_notice("You feel like the last of the [src.name] are out of your system."))
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:2918]:
```suggestion
	to_chat(living_anthill, span_notice("You feel like the last of the [name] are out of your system."))
```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 4]
```diff
@@ -2936,14 +2942,28 @@
...
 		return
 
-	var/obj/effect/decal/cleanable/ants/pests = exposed_turf.spawn_unique_cleanable(/obj/effect/decal/cleanable/ants)
+	var/obj/effect/decal/cleanable/ants/pests = exposed_turf.spawn_unique_cleanable(ants_decal)
 	if(!pests)
 		return
 
 	var/spilled_ants = (round(reac_volume,1) - 5) // To account for ant decals giving 3-5 ants on initialize.
-	pests.reagents.add_reagent(/datum/reagent/ants, spilled_ants)
+	pests.reagents.add_reagent(src.type, spilled_ants)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:2950]:
```suggestion
	pests.reagents.add_reagent(type, spilled_ants)
```

Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm:2950]: the argument is `reagent_type` here so you're safe for just `type`, had to double check

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 5]
```diff
@@ -2936,14 +2942,28 @@
...
 	pests.update_ant_damage()
 
+/datum/reagent/ants/fire
+	name = "Fire ants"
+	description = "A rare mutation of space ants, born from the heat of a plasma fire. Their bites land a 3.7 on the Schmidt Pain Scale."
+	color = "#b51f1f"
+	taste_description = "tiny flaming legs scuttling down the back of your throat"
+	ant_damage_coeff = 0.05 // Roughly 64 brute with 100u
+	ants_decal = /obj/effect/decal/cleanable/ants/fire
+	status_effect = /datum/status_effect/ants/fire
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:2960]: thoughts on pH changes?

Comment (Momo8289) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm:2960]: Not against it, just didn't really see a reason for the pH to be different.

Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm:2960]: I guess being spicy doesnt necessarily mean pH so wouldnt be a cute insert there

File: code/datums/status_effects/debuffs/debuffs.dm [hunk 6]
```diff
@@ -868,7 +870,7 @@
 	if(need_mob_update)
 		victim.updatehealth()
 	if(ants_remaining <= 0 || victim.stat >= HARD_CRIT)
-		victim.remove_status_effect(/datum/status_effect/ants) //If this person has no more ants on them or are dead, they are no longer affected.
+		victim.remove_status_effect(src.type) //If this person has no more ants on them or are dead, they are no longer affected.
```
Comment (JohnFulpWillard) [inline code/datums/status_effects/debuffs/debuffs.dm:873]:
```suggestion
		victim.remove_status_effect(type) //If this person has no more ants on them or are dead, they are no longer affected.
```

File: code/datums/status_effects/debuffs/debuffs.dm [hunk 7]
```diff
@@ -792,6 +792,8 @@
 	processing_speed = STATUS_EFFECT_NORMAL_PROCESS
 	/// Will act as the main timer as well as changing how much damage the ants do.
 	var/ants_remaining = 0
+	/// Multiplied by the number of ants whil calculating damage
+	var/ant_damage_coeff = 0.0016
```
Comment (LemonInTheDark) [inline code/datums/status_effects/debuffs/debuffs.dm:796]: should just call this damage_per_ant. 
also uh, this comment isn't all that useful but I'm not sure there's a better one

File: code/datums/status_effects/debuffs/debuffs.dm (hunk 6 above)
Comment (LemonInTheDark) [inline code/datums/status_effects/debuffs/debuffs.dm:873]: you do not need to use src.type here, just use type

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 8]
```diff
@@ -2874,8 +2874,14 @@
 	ph = 4.6 // Ants contain Formic Acid
 	/// How much damage the ants are going to be doing (rises with each tick the ants are in someone's body)
 	var/ant_damage = 0
+	/// Multiplied by ant_damage to find the damage dealt
+	var/ant_damage_coeff = 0.025
```
Comment (LemonInTheDark) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:2878]: ditto here. also this comment is more like, how it's used currently rather then what it represents, which makes it at best redundant and at worse makes it a lie. comments like this should describe intent rather then the exact process

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 8 above)
Comment (Momo8289) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: Updated this and the other one, hopefully a bit better now

**Comment by Momo8289** [comment]
> LemonInTheDark added the Verified label

:/

**Review by Ghommie — APPROVED** [review]

**Pull request merged**